### PR TITLE
Real data

### PR DESCRIPTION
**Pull request by Rebecca-Rumney**
Some colourful graphs with real data.

### CODE REVIEW

File: PKPD/gui/simulation.py [hunk 1]
```diff
@@ -24,70 +26,114 @@ def __init__(self, main_window):
         self.enable_line_removal = False
         self.is_single_output_model = True
         self.parameter_values = None
+        self.patients_data = None
+        self.patients_dose = None
+        self.state_dimension = 0
+        self.first_patient_id = None
+
+        # 30 colours from https://medialab.github.io/iwanthue/
```
Comment (simonmarchant) [inline PKPD/gui/simulation.py:34]: yay :tada:

File: PKPD/gui/simulation.py [hunk 2]
```diff
@@ -24,70 +26,114 @@ def __init__(self, main_window):
...
-
     def add_data_to_data_model_plot(self):
         """Adds the data from the in the home tab chosen data file to the previously initialised figure. For multi-
         dimensional data, the figure is split into subplots.
         """
         # load data
         data = pd.read_csv(self.main_window.data_file)
-        time_label, state_labels = data.keys()[0], data.keys()[1:]
+        if data.keys()[0] != 'ID':
+            data.insert(0, 'ID', 1)
```
Comment (simonmarchant) [inline PKPD/gui/simulation.py:63]: Could you document that this is adding an ID column where all results are pt 1?

File: PKPD/gui/simulation.py [hunk 3]
```diff
@@ -24,70 +26,114 @@ def __init__(self, main_window):
...
 
         # set the layout
         layout = QtWidgets.QHBoxLayout()
         layout.addWidget(self.canvas)
         layout.addLayout(self._init_plot_infer_model_group())
         self.setLayout(layout)
 
-
     def add_data_to_data_model_plot(self):
         """Adds the data from the in the home tab chosen data file to the previously initialised figure. For multi-
```
Comment (simonmarchant) [inline PKPD/gui/simulation.py]: Could you update the docstring to explain how it handles multiple subjects?

File: PKPD/gui/mainWindow.py [hunk 4]
```diff
@@ -35,18 +36,18 @@ def __init__(self, app):
...
 
-
     def _set_window_size(self):
         """Keeps an aspect ratio width / height of 5/4 and scales the width such that 0.75 of the screen width is covered. If this
         leads to a window height exceeding the screen height, the aspect ratio is kept and the window height is set to the screen
         height.
         """
-        width_coverage = 0.75 # subjective aesthetical choice
-        aspect_ratio = 5 / 4 # subjective aesthetical choice
+        width_coverage = 0.75  # subjective aesthetical choice
```
Comment (simonmarchant) [inline PKPD/gui/mainWindow.py:44]: I know that this isn't your change, but "aesthetical" is not a word. (it's "aesthetic")

**Review by simonmarchant — CHANGES_REQUESTED** [review]
Only minor requested changes but it outputs this plot (test conditions: 3_comp_MultiOutput_fake_real.csv and 3_bolus_linear.mmt):-

image

File: PKPD/gui/simulation.py (hunk 2 above)
Comment (Rebecca-Rumney) [reply PKPD/gui/simulation.py:63]: Done.

File: PKPD/gui/simulation.py (hunk 3 above)
Comment (Rebecca-Rumney) [reply PKPD/gui/simulation.py]: Done.

File: PKPD/gui/mainWindow.py (hunk 4 above)
Comment (Rebecca-Rumney) [reply PKPD/gui/mainWindow.py:44]: I think at some point next week, I'll go through the code and correct all the spelling mistakes.

Comment (Rebecca-Rumney) [reply PKPD/gui/mainWindow.py:44]: But it's not really necessary right now, as long as you can understand what the code is doing.

File: PKPD/gui/simulation.py [hunk 5]
```diff
@@ -24,70 +26,126 @@ def __init__(self, main_window):
...
+                    print(patient_id)
+                    label = 'patient ' + str(patient_id)
+                    colour = int(len(self.colours) * (patient_id - self.first_patient_id) / len(self.patients_data))
+                    self.data_model_ax[dim].scatter(x=self.patients_data[patient_id][0],
+                                                    y=self.patients_data[patient_id][1][:, dim], label=label,
+                                                    marker='o',
+                                                    color=self.colours[colour],
+                                                    edgecolor='black', alpha=0.5)
                 self.data_model_ax[dim].set_ylabel(state_labels[dim])
                 self.data_model_ax[dim].legend()
```
Comment (simonmarchant) [inline PKPD/gui/simulation.py:143]: If you set this to just plot the legend for the first subplot, it should get round the issues with legends all over the place. You can also set the legend to be off on one side with the loc argument (sets where in fig) and bbox arguments (sets exact location of top left of legend), e.g.:
`legend(loc = 'upper right', bbox_to_anchor=(0.5, 0.5))`

**Pull request closed without merging**